FWCore/Utilities/src/ObjectWithDict.cc

```diff
@@ -74,9 +76,13 @@ namespace edm {
     }
 
     if (to.hasBase(from)) { // down cast
+#ifndef _LIBCPP_VERSION
       // use the internal dynamic casting of the compiler (e.g. libstdc++.so)
       void* address = abi::__dynamic_cast(address_, static_cast<abi::__class_type_info const*>(&from.typeInfo()), static_cast<abi::__class_type_info const*>(&to.typeInfo()), -1);
```

Comment:
This is needed to handle case of multiple inheritance where the pointer to the base and pointer to the derived are not the same. That is unlikely in this case (which is why in the distant pass we just assumed they were the same).
Unfortunately, standard C++ doesn't offer anything like this.